source/common/common/fancy_logger.h

```diff
@@ -94,7 +94,7 @@ FancyContext& getFancyContext();
     static std::atomic<spdlog::logger*> flogger{0};                                                \
     spdlog::logger* local_flogger = flogger.load(std::memory_order_relaxed);                       \
     if (!local_flogger) {                                                                          \
-      getFancyContext().initFancyLogger(FANCY_KEY, flogger);                                       \
+      ::Envoy::getFancyContext().initFancyLogger(FANCY_KEY, flogger);                              \
```

Comment:
I just approved https://github.com/envoyproxy/envoy/pull/12706 to unblock Nighthawk; this was needed for them to update their Envoy dependency, rather than for explicit use in the project.

I think for (2), there will usually be coarse grained control.

I think @penguingao agreed to take on (1)

Reply:
Yep. I'll take on (1).

For (2), I think it was that ENVOY_LOG does not expand on non-Loggable classes, but the new logger macro works anywhere.

I can see two ways forward if we want to merge the two logging macros:
- Expand ENVOY_LOG conditionally based on if the class is Loggable to resolve the expansion problem. The question is that, do we want ENVOY_LOG to be switchable to use fine grained logging?
- Just replace ENVOY_LOG with the fine grained logger. This was the initial plan, but we realized that a global switch of the logger format may break downstream log analyzers and automations.

I think the first option is the way to go, but would like to hear more opinions.

Reply:
Yes +1 on option (1). I think today we support direct logging anyway for ENVOY_LOG_MISC (?) without any Loggable.

My personal preference would be to ultimately unify on a single MACRO and then depending on build/config/etc. settings the user can configure what they want? I mainly just think we should avoid multiple macros as I have no idea why someone would use one or the other and we would need to have guidance on that.

Reply:
SGTM. I'll start to make those changes. Ah, ENVOY_LOG_MISC is almost what we needed, just need to get the fine grain logger through similar mechanism.